src/signalrclient/connection_impl.cpp

```diff
@@ -212,6 +212,18 @@ namespace signalr
                             .append(ex.what()));
                     }
                 }
+                try
+                {
+                    connection->m_disconnect_cts->cancel();
```

Comment:
Does this fire the `set_disconnected` callback? I would hope not since `start` never succeeded. Can we verify this in the test?

Reply:
It does not, updated the test